internal/version_test.go

```diff
@@ -46,47 +47,39 @@ func TestKernelVersion(t *testing.T) {
 	}
 }
 
-func TestVersionDetection(t *testing.T) {
+func TestCurrentKernelVersion(t *testing.T) {
+	v, err := KernelVersion()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if evStr := os.Getenv("KERNEL_VERSION"); evStr != "" {
+		ev, err := NewVersion(evStr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if ev[0] != v[0] || ev[1] != v[1] {
+			t.Errorf("expected kernel version %d.%d, got %d.%d", ev[0], ev[1], v[0], v[1])
+		}
+	}
+}
+
+func TestVersionFromCode(t *testing.T) {
 	var tests = []struct {
 		name string
-		s    string
+		code uint32
 		v    Version
-		err  bool
 	}{
-		{"ubuntu version_signature", "Ubuntu 4.15.0-91.92-generic 4.15.18", Version{4, 15, 18}, false},
```

Comment:
So these "weird" kernels export which version in the vDSO? `4.15.0` or `4.15.18`?

Reply:
`4.15.18` would be exported